g2o/types/slam2d/edge_se2_pointxy_bearing.cpp

```diff
@@ -52,6 +52,26 @@ void EdgeSE2PointXYBearing::initialEstimate(
   l2->setEstimate(t * vr);
 }
 
+#ifndef NUMERIC_JACOBIAN_TWO_D_TYPES
+void EdgeSE2PointXYBearing::linearizeOplus() {
+  const VertexSE2* vi = static_cast<const VertexSE2*>(_vertices[0]);
+  const VertexPointXY* vj = static_cast<const VertexPointXY*>(_vertices[1]);
+  const double& x1 = vi->estimate().translation()[0];
+  const double& y1 = vi->estimate().translation()[1];
+  const double& x2 = vj->estimate()[0];
+  const double& y2 = vj->estimate()[1];
+
+  double aux = (std::pow(x2 - x1, 2) + std::pow(y2 - y1, 2));
+
+  _jacobianOplusXi(0, 0) = (y1 - y2) / aux;
+  _jacobianOplusXi(0, 1) = (x2 - x1) / aux;
```

Comment:
@RainerKuemmerle what is the correct value to return when the Jacobian is undefined, i.e. NaN?

Reply:
As the error function relies on atan2(y,x) which is undefined for x=0 && y=0 I would for now leave as is.